README.md

```diff
@@ -101,3 +101,28 @@ while True:
 
 In this case on each iteration the library checks for a new version as described above and resets the device if a new version is available. After the reset
 the `ota_update`-method called in the boot.py performs the actual update.
+
+## HTTP(S) Basic Authentication
+
+`ota_update()` and `check_for_ota_update()` methods allow an optional `auth` parameter.  When specified, `auth` is the base64 encoded `username:password` string (e.g. if username is `foo` and password is `bar`, the base64 encoding of `foo:bar` is `Zm9vOmJhcgo=`).  Use of HTTPS (versus HTTP) is very highly recommended when using `auth` as, otherwise, the resulting username/password are sent in the clear i.e. completely unsecure.
+
+Here is the same example as above, but using HTTPS and Basic Authentication:
+
+```python
+import micropython_ota
+
+# connect to network
+
+ota_host = 'https://example.com'
+project_name = 'sample'
+filenames = ['boot.py', 'main.py']
+userpass_b64 = 'Zm9vOmJhcgo='   # it's best to place this credential is a secrets.py file
```

Comment:
What about providing the username and password and let Python do the base64-magic?